app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute
+        authorize! sample.project, to: :update_sample?
+
+        validate_sample_in_project
+
+        validate_metadata_param
+
+        transform_metadata_keys
+
+        metadata_fields_update_status = metadata_modification
+
+        @sample.save
+
+        fields_not_updated(metadata_fields_update_status)
+        metadata_fields_update_status
+      rescue Samples::Metadata::UpdateService::SampleMetadataUpdateError => e
+        @sample.errors.add(:base, e.message)
+        metadata_fields_update_status || nil
+      end
+
+      private
+
+      def validate_sample_in_project
+        return unless @project.id != @sample.project.id
+
+        raise SampleMetadataUpdateError,
+              I18n.t('services.samples.metadata.sample_does_not_belong_to_project', sample_name: @sample.name,
+                                                                                    project_name: @project.name)
+      end
+
+      def validate_metadata_param
+        return unless @metadata.nil? || @metadata == {}
+
+        raise SampleMetadataUpdateError,
+              I18n.t('services.samples.metadata.empty_metadata', sample_name: @sample.name)
+      end
+
+      def transform_metadata_keys
+        # Without transforming keys, issues with overwritting can occur and multiples of the same key can appear
+        @metadata = @metadata.transform_keys(&:to_s)
+      end
+
+      def metadata_modification
+        update_status = { updated: [], not_updated: [] }
+        @metadata.each do |key, value|
+          if value.blank?
+            if @sample['metadata'].key?(key)
```

Comment:
**nitpick (blocking):** can you change all instances of `@sample['metadata']` to `@sample.metadata` as that is the normal way to access a models attributes vs the hash like interface your using.

Reply:
Fixed in 38058670dd23177ebf0267bb3f1c3fa5166647d9